src/main/scala/mesosphere/marathon/core/group/impl/GroupManagerImpl.scala

```diff
@@ -208,11 +212,33 @@ class GroupManagerImpl(
 
   @SuppressWarnings(Array("all")) // async/await
   override def invalidateGroupCache(): Future[Done] = async {
+    root := None
+
     // propagation of reset group caches on repository is needed,
     // because manager and repository are holding own caches
     await(groupRepository.invalidateGroupCache())
     val currentRoot = await(groupRepository.root())
     root := Option(currentRoot)
+
+    registerMetrics()
```

Comment:
Why do we have two calls to registerMetrics? Can we have invalidateGroupCache just clear the rootGroup, and then call rootGroup() to repopulate?

Reply:
I did it this way In order to not worry about the order (which one gets invoked first). For instance, right now when Marathon becomes a leader it calls `invalidateGroupCache` twice. Also, with the current implementation `root` is set to `Some(...)` here first. But it may change in the future. So, whoever gets called first, registers metrics.

Reply:
This won't work?

```diff
diff --git a/src/main/scala/mesosphere/marathon/core/group/impl/GroupManagerImpl.scala b/src/main/scala/mesosphere/marathon/core/group/impl/GroupManagerImpl.scala
index 82a74f95d..a7dbd5864 100644
--- a/src/main/scala/mesosphere/marathon/core/group/impl/GroupManagerImpl.scala
+++ b/src/main/scala/mesosphere/marathon/core/group/impl/GroupManagerImpl.scala
@@ -217,10 +217,7 @@ class GroupManagerImpl(
     // propagation of reset group caches on repository is needed,
     // because manager and repository are holding own caches
     await(groupRepository.invalidateGroupCache())
-    val currentRoot = await(groupRepository.root())
-    root := Option(currentRoot)
-
-    registerMetrics()
+    rootGroup()
     Done
   }
 
```

Reply:
I think not, because `groupRepository.root()` has to be invoked.